### ISSUE TITLE
Out-of-class definition of member of constrained specialization

### ISSUE THREAD

**Issue by jwakely** [llvm/llvm-project#58124]
```c++
template<class T> concept C1 = true;
template<class T> concept C2 = C1<T> && true;

template<C1> struct X { };
template<C2 T> struct X<T> { struct Y; };

template<C2 T> struct X<T>::Y { }; // (1)
```

Clang rejects (1) saying:

```console
<source>:7:10: error: type constraint differs in template redeclaration
template<C2 T> struct X<T>::Y { };
         ^
<source>:4:10: note: previous template declaration is here
template<C1 T> struct X { };
         ^
<source>:7:29: error: no struct named 'Y' in 'X<T>'
template<C2 T> struct X<T>::Y { };
                      ~~~~~~^
```

This is wrong, `Y` is a member of the partial specialization on line 5, not the primary template on line 4.

**Comment by royjacobson** [comment]
Looks like duplicate of #50208.

**Comment by jwakely** [comment]
#50208 is now closed as a dup of #49620

**Comment by usx95** [comment]
Duplicate of #49620
